File: ClassPrevious/class/99/Hw3/whiteb2/Actor.java
//////////////////////////////////////////////////////////////////////////
//
//  This code is for instructional purposes only. It was generated for
//  use in a graduate level course to show certain aspects of data
//  storage algorithms. It has problems and should not be used
//  outside the class environment.
//
//  Author:  Thomas D. Citriniti     citrit@rpi.edu
//  Class:   Advanced Computer Graphics and Data visualization
//           Rensselaer Polytechnic Institute
//  Date:    January 1998
//
/////////////////////////////////////////////////////////////////////////

import java.util.Vector;
import jogl.*;

/**
 * Actor class, this class holds a collection of cells and properties.
 *
 * @author		Thomas D. Citrinit
 * @version		0.1
 */

// Display lists added by Bryan Whitehead
public class Actor extends ParentObject
{

	/**
	  * Collection of Cells that makes up an entity
	  */
	
	Vector cells;
	int DLnumber;
	/** Orientation as rotations around X,Y, and Z */
	float orientation[];
	PointType position; // Obvious
	float scale[]; // Obvious
	JoglCanvas gl;


	/** Constructor that creates a Vector class for cells */
	public Actor() 
	{
		cells = new Vector();
		orientation = new float[3];
		position = new PointType();
		scale = new float[3];
		scale[0] = scale[1] = scale[2] = 1;
		DLnumber=-1;
	}

	public void beginDL(int DL){
		int n;
		if(gl.isList(DL))
		{
			gl.deleteLists(DL, 1);
			n = generateDL();
			DL = n;
		}		
		gl.newList(DL, GL.COMPILE);
	}
	public void endDL(){
		gl.endList();
	}
					
	public void callDL(int DL){
		gl.callList(DL);
	}
	
	public int generateDL(){
		return gl.genLists(1000);
	}

	
	/** Method called by Renderer to signal an update is needed. */
	public void render(Renderer aren)
	{
		if(DLnumber>-1) {
			callDL(DLnumber);
		}
		else {
			DLnumber=generateDL();
			beginDL(DLnumber);
			aren.pushModelMatrix();
			aren.setOrientation(orientation, scale, position);
			for (int i=0;i<cells.size();i++) {
				((Cell)cells.elementAt(i)).render(aren);
			}
			aren.popModelMatrix();
			endDL();
		}
	}

	/** Change orientation */
	public void rotateX(float angle)
	{
		orientation[0] += angle;
	}
	/** Change orientation */
	public void rotateY(float angle)
	{
		orientation[1] += angle;
	}
	/** Change orientation */
	public void rotatez(float angle)
	{
		orientation[2] += angle;
	}
	public void scale(float sc[])
	{
		scale[0] = sc[0]; scale[1] = sc[1]; scale[2] = sc[2];
	}
	public void translate(float x, float y, float z)
	{
		position.x += x;
		position.y += y;
		position.z += z;
	}
	public void translate(double x, double y, double z)
	{
		position.x += x;
		position.y += y;
		position.z += z;
	}

	/** Add Cells to the actors collection. */
	public void addCell(Cell cell) { cells.addElement(cell); }
}

File: Classes/code/objRobot/OGLRen/Actor.js
//////////////////////////////////////////////////////////////////////////
//
//  This code is for instructional purposes only. It was generated for
//  use in a college course to show certain aspects of data
//  storage algorithms. It has problems and should not be used
//  outside the class environment.
//
//  Author:  Thomas D. Citriniti     citrit@rpi.edu
//  Class:   Computer Graphics 
//           Rensselaer Polytechnic Institute
//  Date:    Sept 2015
//
//////////////////////////////////////////////////////////////////////////

var OGLRen = OGLRen || {};

/**
  * Actor class is a holder of cells into single objects.
  */
OGLRen.Actor = function() {
    var cells = [];
    /** Orientation as rotations around X,Y, and Z */
	var orientation = [0.0, 0.0, 0.0];
	var position = [0,0,0]; // Obvious
	var scale = [1.0, 1.0, 1.0]; // Obvious
    
    this.copy = function() {
        var cp = new OGLRen.Actor();
        
    }
    
    /** Method called by Renderer to signal an update is needed. */
	this.render = function( aren)
	{
		aren.pushModelMatrix();
		aren.setOrientation(orientation, scale, position);
		for (var i=0;i<cells.length;i++) {
			cells[i].render(aren);
		}
		aren.popModelMatrix();
	}

	/** Change orientation */
	this.rotateX = function( angle)
	{
		orientation[0] += angle;
	}
	/** Change orientation */
	this.rotateY = function( angle)
	{
		orientation[1] += angle;
	}
	/** Change orientation */
	this.rotateZ = function( angle)
	{
		orientation[2] += angle;
	}
	this.scale = function( sc)
	{
		scale[0] = sc[0]; scale[1] = sc[1]; scale[2] = sc[2];
	}
	this.translate = function( x,  y,  z)
	{
		position[0] += x;
		position[1] += y;
		position[2] += z;
	}
	
	/** Add Cells to the actors collection. */
	this.addCell = function( cell) { cells.push(cell); }
    
}
OGLRen.Actor.inheritsFrom(OGLRen.ParentObject);
